conans/model/build_info.py

```diff
@@ -50,41 +116,55 @@ def _filter_paths(self, paths):
         else:
             return abs_paths
 
+    def _get_paths(self, path_name):
+        """
+        Get the absolute paths either composing the lists from components or from the global
+        variables. Also filter the values checking if the folders exist or not and avoid repeated
+        values. The paths are calculated once and then the result is cached.
+        :param path_name: name of the path variable to get (include_paths, res_paths...)
+        :return: List of absolute paths
+        """
+        def get_paths_value():
+            return getattr(self, "_%s_paths" % path_name)
+
+        if get_paths_value() is None:
+            if self._deps:
+                self.__dict__["_%s_paths" % path_name] = []
+                for dep_value in self._deps.values():
+                    abs_paths = self._filter_paths(getattr(dep_value, "%s_paths" % path_name))
+                    if not get_paths_value():
+                        self.__dict__["_%s_paths" % path_name].extend(abs_paths)
+                    for path in abs_paths:
+                        if path not in get_paths_value():
+                            self.__dict__["_%s_paths" % path_name].append(path)
+            else:
+                abs_paths = self._filter_paths(getattr(self, "%sdirs" % path_name))
+                self.__dict__["_%s_paths" % path_name] = abs_paths
+        return get_paths_value()
```

Comment:
It would be more easy to return just the list of `abs_paths` from this function and, if you want to add cache, you can add it into the properties:

```python
@proprty
def include_paths(self):
    if not self._include_paths:
         self._include_paths = self._get_paths("include")
    return self._include_paths
```

...or we can write a `@property_cached` decorator to avoid repeating the same pattern again and again.